pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -271,17 +271,86 @@ def _on_evaluation_epoch_end(self) -> None:
         self.trainer._call_lightning_module_hook("on_epoch_end")
         self.trainer.logger_connector.on_epoch_end()
 
+    def _get_keys(self, iterable: Union[List, dict]) -> Iterable[str]:
```

Comment:
You might find this function interesting to apply a function to a collection, filtering by type.

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/apply_func.py#L71

I didn't look to deep into the actual implementation yet but looks like you could use it instead of these 2 you wrote here

Reply:
Will take a look at this!

Reply:
I've given it a first try as follows:
```
apply_to_collection(new_results, Dict, dict.keys)
```
to get the keys, and:
```
apply_to_collection(metrics, Dict, dict.get, metric[0])
```
to get the values, but there seems to be a problem with the recursion as in the following example:
```
self.log("performance", {"acc": loss, "recall": loss})
```
I don't get 'acc' and 'recall' back, but only 'performance' via dict.keys, and the values also do not get returned via dict.get.

I will look further into this, but if you have any ideas, please share them!